### PR TITLE
ci: Add GitHub CI job to check Markdown links during PRs

### PR DESCRIPTION
**Pull request by Rei1010**
Signed-off-by: wen.rui <wen.rui@daocloud.io>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Verify all markdown links correctly worked in modified files

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by Rei1010** [comment]
@fisherxu @gy95  sorry for accidentally closed previous PR https://github.com/kubeedge/kubeedge/pull/4139.
Variable"use-quiet-mode" will show all check results if set to "no", "use-verbose-mode" will show status code is set to "yes", "check-modified-files-only" only check modified files if set to "yes"
You can check all variables here: [gaurav](https://github.com/gaurav-nelson/github-action-markdown-link-check#custom-variables)

**Comment by Rei1010** [comment]
Attached CI result in my fork: https://github.com/Rei1010/kubeedge/runs/8034239111?check_suite_focus=true


=========================> MARKDOWN LINK CHECK <=========================

FILE: .github/workflows/test.md
[✓] https://github.com/kubeedge/kubeedge → Status: 200
[✖] https://github.com/kubeedge/kubefake → Status: 404

2 links checked.

ERROR: 1 dead links found!
[✖] https://github.com/kubeedge/kubefake → Status: 404

=========================================================================

**Comment by gy95** [comment]
this workflow didn't find existing broken links?

**Comment by Rei1010** [comment]
> this workflow didn't find existing broken links?

@gy95 variable "check-modified-files-only" set to "no" will check all links under markdown file

**Comment by gy95** [comment]
> > this workflow didn't find existing broken links?
> 
> @gy95 variable "check-modified-files-only" set to "no" will check all links under markdown file

I mean we can temporarily set  "check-modified-files-only" to "no", and fix the exist broken links in markdown files, and after fixing those broken links, we can set it to yes finally

**Comment by Rei1010** [comment]
> > > this workflow didn't find existing broken links?
> > 
> > 
> > @gy95 variable "check-modified-files-only" set to "no" will check all links under markdown file
> 
> I mean we can temporarily set "check-modified-files-only" to "no", and fix the exist broken links in markdown files, and after fixing those broken links, we can set it to yes finally

@gy95 updated, test result updated in https://github.com/kubeedge/kubeedge/pull/4142#issuecomment-1229133785

File: .mlc_config.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      "urls": ["https://github.com/", "https://guides.github.com/", "https://help.github.com/", "https://docs.github.com/"],
+      "headers": {
+        "Accept-Encoding": "zstd, br, gzip, deflate"
+      }
+    }
+  ],
+  "ignorePatterns": [
+    {
+      "pattern": "^https://github.com/kubeedge/kubeedge/issues/"
+    },
```
Comment (gy95) [inline .mlc_config.json:18]: what is this used for? i thought this is no need.

File: docs/proposals/device-crd.md [hunk 2]
```diff
@@ -761,11 +761,11 @@ IoT device lifecycle management comprises of several steps listed below
   - If a device is damaged , it needs to be retired. This is currently not in scope of this design.
 
 ### Device Actions
-- Currently the device model doesn't support [WOT style actions](https://iot.mozilla.org/wot/#actions-resource). The only way to perform some action on the device is by changing the desired state of the twin property in the status field. We need to see how WOT actions can be incorporated in this model and who will consume those actions ?
+- Currently the device model doesn't support [WOT style actions](https://webthings.io/wot/#actions-resource). The only way to perform some action on the device is by changing the desired state of the twin property in the status field. We need to see how WOT actions can be incorporated in this model and who will consume those actions ?
```
Comment (gy95) [inline docs/proposals/device-crd.md:764]: https://webthings.io/wot/#actions-resource is still a broken link

File: docs/proposals/device-crd.md [hunk 3]
```diff
@@ -761,11 +761,11 @@ IoT device lifecycle management comprises of several steps listed below
...
 Will it be the responsibility of the mapper to expose HTTP APIs for the actions ? Can we generate server / client code to perform / consume these actions ? Can we handle firmware updates with such actions ?
 
 ### Device Events
-- Currently the device model doesn't support  [WOT style events](https://iot.mozilla.org/wot/#events-resource). Should telemetry data like sensor temperature readings etc emitted by devices be reported as events ? WOT events are some abnormal events reported by the device like overheated, or reboot etc.
+- Currently the device model doesn't support  [WOT style events](https://webthings.io/wot/#events-resource). Should telemetry data like sensor temperature readings etc emitted by devices be reported as events ? WOT events are some abnormal events reported by the device like overheated, or reboot etc.
```
Comment (gy95) [inline docs/proposals/device-crd.md:768]: ditto

File: docs/proposals/edgemesh-design.md [hunk 4]
```diff
@@ -46,7 +46,7 @@ The cloud native and microservice architecture is becoming more and more popular
 * Application access service with the same mechanism no matter it located on cloud or edge
 
 ### Service discovery on edge
-* Use iptables to redirect inbound and outbound traffic to the EdgeHub/ServiceBus. Then do service discovery in EdgeHub/ServiceBus by pick up the right POD from the service definition. The idea is similar with what istio is doing in the init_container: https://github.com/istio/istio/blob/master/tools/packaging/common/istio-iptables.sh
+* Use iptables to redirect inbound and outbound traffic to the EdgeHub/ServiceBus. Then do service discovery in EdgeHub/ServiceBus by pick up the right POD from the service definition. The idea is similar with what istio is doing in the init_container: https://github.com/istio/istio/blob/032be5b1ecfd5b087f623d76dca2879f8cd130be/tools/packaging/common/istio-iptables.sh
```
Comment (gy95) [inline docs/proposals/edgemesh-design.md:49]: how do you find this link `https://github.com/istio/istio/blob/032be5b1ecfd5b087f623d76dca2879f8cd130be/tools/packaging/common/istio-iptables.sh`, is this what action find or recommend?

Comment (Rei1010) [reply docs/proposals/edgemesh-design.md:49]: No, istio-iptables.sh is removed in https://github.com/istio/istio/pull/19273, the link is the last work path.

File: docs/proposals/device-crd.md (hunk 2 above)
Comment (Rei1010) [reply docs/proposals/device-crd.md:764]: I can't find the correctly link in webthings, any idea about this?

File: .mlc_config.json (hunk 1 above)
Comment (Rei1010) [reply .mlc_config.json:18]: I will remove it.

File: docs/proposals/device-crd.md (hunk 2 above)
Comment (JiaweiGithub) [reply docs/proposals/device-crd.md:764]: > I can't find the correctly link in webthings, any idea about this?
I changed it and it has been merged

**Review by wlq1212 — APPROVED** [review]
/lgtm

**Pull request closed without merging**